driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -581,9 +583,9 @@ boolean setPool(HostConnectionPool pool) {
      * The connection should generally not be reused after that.
      */
     void release() {
-        HostConnectionPool pool = poolRef.get();
-        if (pool != null)
-            pool.returnConnection(this);
+        Owner owner = ownerRef.get();
+        if (owner instanceof HostConnectionPool)
```

Comment:
Hrmm, maybe I'm wrong about how this goes about, but I do see that inFlight is not reduced on connection close.  However, returnConnection would eventually be closed on request timeout/cancel, so maybe this isn't a big deal.